roles/repositories/tasks/main.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: 'Enable Red Hat Repositories'
+  theforeman.foreman.repository_set:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    name: "{{ item.1.name }}"
+    state: enabled
+    repositories:
+      - releasever: "{{ item.1.releasever | default(omit) }}"
+        basearch: "{{ item.1.basearch | default(omit) }}"
+  with_subelements:
+    - "{{ products | selectattr('repository_sets', 'defined') | list }}"
+    - repository_sets
+
+- name: 'Create Products'
+  theforeman.foreman.product:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    name: "{{ item.name }}"
+    state: present
+  with_items:
+    - "{{ products | selectattr('repositories', 'defined') | list }}"
```

Comment:
TODO: Should technically be able to define a Product without defining Repositories on that Product. I'm not sure if there is a real use for that, but Katello allows it and I noticed it would be nice to have when developing test playbooks for other roles.